src/artboard.cpp

```diff
@@ -369,7 +369,15 @@ void Artboard::update(ComponentDirt value)
 	if (hasDirt(value, ComponentDirt::Path))
 	{
 		m_ClipPath->reset();
-		m_ClipPath->addRect(0.0f, 0.0f, width(), height());
+		if (m_FrameOrigin)
+		{
+			m_ClipPath->addRect(0.0f, 0.0f, width(), height());
+		}
+		else
+		{
+			m_ClipPath->addRect(
+			    -width() * originX(), -height() * originY(), width(), height());
+		}
```

Comment:
Can't this be generalized so that `originX()` and `originY()` are `0` when `m_FrameOrigin` is set?

Reply:
We could, but we'd have to virtualize originX() and originY() which our generator doesn't currently support. Also doing it like this saves an unnecessary transform per artboard. But your solution would be more succinct...